src/services/State.ts

```diff
@@ -76,24 +77,21 @@ export class StateService {
     return { first, last };
   }
 
-  setSelected(selected: any[]): StateService {
-    if (!this.selected) {
-      this.selected = selected || [];
-    } else {
-      this.selected.splice(0, this.selected.length);
-      this.selected.push(...selected);
-    }
+  get selected(): any[] {
+    return this.rows.filter(row => this.isRowSelected(row));
```

Comment:
Done in https://github.com/swimlane/angular2-data-table/pull/154/commits/3fb87f80042a23ac3f386ed6480fe7827494263a .